src/Utils/BlockTemplateUtils.php

```diff
@@ -119,12 +118,13 @@ public static function gutenberg_inject_theme_attribute_in_content( $template_co
 
 	/**
 	 * Build a unified template object based a post Object.
+	 * Important: This method is an almost identical duplicate from wp-includes/block-template-utils.php as it was not intended for public use.
```

Comment:
Since it was not intended for public use, perhaps let's add `@internal` here as well? Also, what about mentioning why do we need this if it is “almost” identical to the other one? What do you think?

Reply:
The entire class is marked as `@internal` at the top of the file. I think it would add confusion if we then add it to various other methods as that would insinuate non-marked `@internal` methods are OK for public use?

As for the difference between this and its original counterpart, I will write something non-specific to avoid us having to update this comment with every future change. I think as long as the reader knows its not identical that should be good too.